arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,306 @@
+    let symbolStyle = AGSSymbolStyle(styleName: "Esri2DPointSymbolsStyle", portal: .arcGISOnline(withLoginRequired: false))
+    /// The data source that handles symbols retrieval and for the legend table display.
+    private let symbolsDataSource = SymbolsDataSource()
+    
+    /// A feature layer with LA County Points of Interest service.
+    let featureLayer: AGSFeatureLayer = {
+        let serviceFeatureTable = AGSServiceFeatureTable(url: URL(string: "http://services.arcgis.com/V6ZHFr6zdgNZuVG0/arcgis/rest/services/LA_County_Points_of_Interest/FeatureServer/0")!)
+        let layer = AGSFeatureLayer(featureTable: serviceFeatureTable)
+        return layer
+    }()
+    
+    // MARK: Methods
+    
+    /// Create a map with an `AGSFeatureLayer` added to its operational layers.
+    /// - Parameter featureLayer: An `AGSFeatureLayer` object.
+    /// - Returns: An `AGSMap` object.
+    func makeMap(featureLayer: AGSFeatureLayer) -> AGSMap {
+        let map = AGSMap(basemapStyle: .arcGISLightGray)
+        map.referenceScale = 1e5
+        map.operationalLayers.add(featureLayer)
+        return map
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["CreateSymbolStylesFromWebStylesViewController"]
+        
+        // Get the symbols from the symbol style hosted on an ArcGIS portal.
+        symbolsDataSource.getSymbols(symbolStyle: symbolStyle, categories: SymbolCategory.allCases) { [weak self] in
+            guard let self = self else { return }
+            self.featureLayer.renderer = self.symbolsDataSource.makeUniqueValueRenderer(fieldNames: ["cat2"])
+            self.legendBarButtonItem.isEnabled = true
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        mapScaleObservation = mapView.observe(\.mapScale, options: .initial) { [weak self] (_, _) in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                self.featureLayer.scaleSymbols = self.mapView.mapScale >= 8e4
+            }
+        }
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        mapScaleObservation = nil
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "LegendTableSegue",
+           let controller = segue.destination as? UITableViewController {
+            controller.presentationController?.delegate = self
+            controller.tableView.dataSource = symbolsDataSource
+            // Hold a weak reference to the table view to asynchronously reload
+            // image swatches for the symbols.
+            symbolsDataSource.tableView = controller.tableView
```

Comment:
`UITableViewDiffableDataSource` takes the table view as a parameter to the initializer and then automatically assigns itself as the table view's data source. You might consider doing something similar with `SymbolsDataSource.tableView` so that you don't need both of these.